Migrate to Capsize (#696)
seek-oss/braid-design-system: .changeset/gorgeous-bears-glow.md

Migrating off this prop will require consumers to perform the following steps:
- Remove the usage of `_LEGACY_SPACE_` on a component
- Conduct a visual review of the impact (component will appear closer to neighbouring elements)
- Use existing layout components, e.g. `Stack`, to define/control the reintroduction of the desired space.

Any issues, please reach out to the team as we are happy to help support consumers in migrating.

**Theme Tokens: Text and Heading definitions**

There have been strutural theme changes for `heading` and `text` definitions to support the defining of `capHeight` in the system. A definition will now have either `fontSize` or `capHeight`, eg:

```diff
{
  text: {
    standard: {
      mobile: {
-        size: 16,
+        fontSize: 16, // or `capHeight` if using the new technique.
        rows: 6
      }
    }
  }
}
```

This is likely to not affect consumers, unless using these values off the theme explicitly in custom treat files.

**Theme Tokens: Descender and Cap Height scales**

> Instead of the calculated values of `capHeightScale` and `decenderHeightScale`, a theme now accepts `fontMetrics`—a structure the represents the metadata from the font itself.

Reviewer: ```suggestion
Instead of the calculated values of `capHeightScale` and `decenderHeightScale`, a theme now accepts `fontMetrics`—a structure that represents the metadata from the font itself.
```

---
'braid-design-system': major
---

Improved trimming of white space around text

> ## Breaking Changes

Reviewer: I don't think we can use headings here since this ultimately gets nested inside a bullet list. This is why I usually go for the bold + all-caps style.

> Improved the mechanism used to trim the white space around text. This will have subtle white space changes throughout the UI, but will largely just be improvements on balance of space around text.

Reviewer: Should paragraph be before the "Breaking changes" heading?

> As a result of adopting the new technique it is no longer feasible to maintain the `_LEGACY_SPACE_` fallback on `Heading` and `Text`.

Reviewer: Should this section be a bullet list summary of the sections below?

Migrating off this prop will require consumers to perform the following steps:
- Remove the usage of `_LEGACY_SPACE_` on a component
- Conduct a visual review of the impact (component will appear closer to neighbouring elements)
- Use existing layout components, e.g. `Stack`, to define/control the reintroduction of the desired space.

Any issues, please reach out to the team as we are happy to help support consumers in migrating.

**Theme Tokens: Text and Heading definitions**

There have been strutural theme changes for `heading` and `text` definitions to support the defining of `capHeight` in the system. Previously a definition contained `size` that was the specified font size inclusive of surrounding white space.

A definition now has `capHeight` which is representative of the visual height, supporting improved alignment with other elements like, icons etc.

```diff
{
  text: {
    standard: {
      mobile: {
-        size: 16,
+        capHeight: 11.43,
        rows: 6
      }
    }
  }
}
```

> This is not likely to affect consumers, unless using these theme values explicitly in custom treat files.

Reviewer 2: ```suggestion
This is not likely to affect consumers, unless these theme values are used explicitly in custom treat files.
```

---
'braid-design-system': major
---

> Improved trimming of white space around text

Reviewer: ```suggestion
**`apac` and `seekBusiness` themes:** Improved trimming of white space around text
```

Reviewer: Ignore me, I forgot that this also changes the height of text elements in the other themes.
